### PR TITLE
Custom background change support

### PR DESCRIPTION
**Pull request by ermiyaeskandary**
@tjorim 's work - just speeding it up
**Confirmed fully working personally**

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -1104,6 +1104,13 @@ var userInterface = window.userInterface = (function() {
                     window.log('Automatic Respawning set to: ' + window.autoRespawn);
                     userInterface.savePreference('autoRespawn', window.autoRespawn);
                 }
+                // Letter 'B' to prompt for a custom background url
+                if (e.keyCode === 66) {
+                    var url = prompt("Please enter a background url:");
```
Comment (ermiyaeskandary) [inline bot.user.js]: Well, expect the prompt...

**Comment by Defimatt** [comment]
Is this supposed to work with backgrounds of all sizes? I tested it with a small tile pattern, and it's not scaled / tiled properly:

image

**Comment by ermiyaeskandary** [comment]
@MattDuffin No - I don't think so. It's slither.io's own server code. Can you test the same image with another script which lets you do this? Does it scale fully ?

**Comment by Defimatt** [comment]
@ErmiyaEskandary No, I haven't seen it in any other scripts, I just wondered whether it was supposed to be in there.

After researching the issue, I think the only way to do it would be to intercept the canvas commands.

LGTM.

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/276/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by ChadSki** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/276/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by tjorim** [comment]
:+1: (can be improved over time)

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/276/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by j-c-m** [comment]
This code was not even tested for 1 game, overlay messed up.

image

**Pull request merged**